device/common/include/traccc/edm/device/sort_key.hpp

```diff
@@ -0,0 +1,34 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "traccc/definitions/primitives.hpp"
+#include "traccc/edm/track_parameters.hpp"
+
+namespace traccc::device {
+
+struct sort_key {
+    traccc::scalar key;
+};
```

Comment:
Do we expect the key to become something more complicated than a float value? ("Yes" is absolutely a believable answer.)

If not, it would be easier to just do:

```suggestion
using sort_key = traccc::scalar;
```

This would allow us to not have to define `sort_key_comp`. :thinking:

Reply:
OK I will just use `scalar`. Let's think about making struct if we ever need to do that in the future.